modules/external-secrets/variables.tf

```diff
@@ -36,11 +36,19 @@ variable "addon_context" {
 variable "external_secrets_ssm_parameter_arns" {
   description = "List of Systems Manager Parameter ARNs that contain secrets to mount using External Secrets"
   type        = list(string)
-  default     = []
+
+  validation {
+    condition     = length(var.external_secrets_ssm_parameter_arns) > 0
+    error_message = "Must contain at least parameter store ARN"
+  }
 }
 
 variable "external_secrets_secrets_manager_arns" {
   description = "List of Secrets Manager ARNs that contain secrets to mount using External Secrets"
   type        = list(string)
-  default     = []
```

Comment:
Should the default be `[]` like it was previously, wide open access like the [parent module](https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/pull/86/files#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288eL648), or include a validation rule as in this PR?

Note that an empty list makes the IAM code a lot messier, as IAM policies do not allow empty resource blocks.  When developing this PR, I found the least-impactful change to add `not_resources = ["*"]` to the policy if we were passed an empty list, but that also makes things more difficult to understand (and consequently makes external secrets unable to fetch any AWS secrets)